test/Tests/Services/ISqlExecutorTests.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System.Data;
+
+using NSubstitute;
+
+using NUnit.Framework;
+
+using Xperience.Manager.Services;
+
+namespace Xperience.Manager.Tests.Services
+{
+    /// <summary>
+    /// Tests for <see cref="ISqlExecutor"/>.
+    /// </summary>
+    public class ISqlExecutorTests
+    {
+        private const string COL_1 = "First";
+        private const string COL_2 = "Second";
+        private const string VAL_1 = "Value1";
+        private const string VAL_2 = "Value2";
+        private const string CONN_STRING = "TestConnectionString";
+        private const string VALID_SCRIPT = "EnabledUsersWithAdminAccess.sql";
+        private const string SCRIPT_TEXT = @"SELECT
+	UserID,
+	UserName
+FROM
+	CMS_User
+WHERE
+	UserEnabled = 1
+	AND UserAdministrationAccess = 1";
+
+
+        private readonly IDbCommand command = Substitute.For<IDbCommand>();
+        private readonly IDbConnection connection = Substitute.For<IDbConnection>();
+        private readonly IDbDataParameter parameter = Substitute.For<IDbDataParameter>();
+        private readonly IDataParameterCollection parameterCollection = Substitute.For<IDataParameterCollection>();
+
+
+        [Test]
+        public async Task ExecuteQuery_ValidScript_ExecutesAndReturnsData()
+        {
+            var factory = GetFactory();
+            var sqlExecutor = new SqlExecutor(factory);
+
+            var data = await sqlExecutor.ExecuteQuery(CONN_STRING, VALID_SCRIPT);
+
+            Assert.Multiple(() =>
+            {
+                connection.Received().Open();
+                command.Received().ExecuteReader();
+                Assert.That(command.CommandText, Is.EqualTo(SCRIPT_TEXT));
+                Assert.That(data.Count(), Is.EqualTo(2));
+                Assert.That(data.FirstOrDefault()?.Property(COL_1)?.Value.ToString(), Is.EqualTo(VAL_1));
+                Assert.That(data.FirstOrDefault()?.Property(COL_2)?.Value.ToString(), Is.EqualTo(VAL_2));
+
+            });
+        }
+
+
+        [Test]
+        public async Task ExecuteQuery_InvalidScript_Throws()
+        {
+            var factory = GetFactory();
+            var sqlExecutor = new SqlExecutor(factory);
+            string queryName = "InvalidScript.sql";
+
+            try
+            {
+                await sqlExecutor.ExecuteQuery(CONN_STRING, queryName);
+
+                Assert.Multiple(() =>
+                {
+                    connection.DidNotReceive().Open();
+                    command.DidNotReceive().ExecuteReader();
+                    Assert.Fail($"Resource '{queryName}' not found.");
+                });
+            }
+            catch (Exception) { }
```

Comment:
This is dangerous as the Assert.Multiple throws an exception in case of failure. 
It means that assert fail won't fail the test as the exception is caught.
Use Assert.Throws or Assert.ThrowsAsync instead.